DashAI/back/dataloaders/classes/dashai_dataset.py

```diff
@@ -217,6 +226,169 @@ def save_dataset(datasetdict: DatasetDict, path: str) -> None:
+        where the size of the test-validation set is now considered to be 100%.
+        Therefore, the sizes of the test and validation sets will now be
+        calculated as 100%, i.e. as val_size/(test_size+val_size) and
+        test_size/(test_size+val_size) respectively.
+
+    Example:
+
+    If we split a dataset into 0.8 training, a 0.1 test, and a 0.1 validation,
+    in the first process we split the training data with 80% of the data, and
+    the test-validation data with the remaining 20%; and then in the second
+    process we split this 20% into 50% test and 50% validation.
+
+    Parameters
+    ----------
+    total_rows : int
+        Size of the Dataset.
+    train_size : float
+        Proportion of the dataset for train split (in 0-1).
+    test_size : float
+        Proportion of the dataset for test split (in 0-1).
+    val_size : float
+        Proportion of the dataset for validation split (in 0-1).
+    seed : Union[int, None], optional
+        Set seed to control to enable replicability, by default None
+    shuffle : bool, optional
+        If True, the data will be shuffled when splitting the dataset,
+        by default True.
+
+    Returns
+    -------
+    Tuple
+        Train, Test and Validation indices.
+    """
+
+    # Generate shuffled indices
+    np.random.seed(seed)
+    indices = np.arange(total_rows)
+
+    test_val = test_size + val_size
+    val_proportion = test_size / test_val
+    train_indices, test_val_indices = train_test_split(
+        indices,
+        train_size=train_size,
+        random_state=seed,
+        shuffle=shuffle,
+    )
+    test_indices, val_indices = train_test_split(
+        test_val_indices,
+        train_size=val_proportion,
+        random_state=seed,
+        shuffle=shuffle,
+    )
+    return list(train_indices), list(test_indices), list(val_indices)
+
+
+@beartype
+def split_dataset(
+    dataset: Dataset,
+    train_indices: List,
+    test_indices: List,
+    val_indices: List,
+) -> DatasetDict:
+    """Split the dataset in train, test and validation subsets.
+
+    Parameters
+    ----------
+    dataset : DatasetDict
+        A HuggingFace DatasetDict containing the dataset to be split.
+    train_indices : List
+        Train split indices.
+    test_indices : List
+        Test split indices.
+    val_indices : List
+        Validation split indices.
+
+
+    Returns
+    -------
+    DatasetDict
+        The split dataset.
+    """
+
+    # Get the number of records
+    n = len(dataset)
+
+    # Convert the indices into boolean masks
+    train_mask = np.isin(np.arange(n), train_indices)
+    test_mask = np.isin(np.arange(n), test_indices)
+    val_mask = np.isin(np.arange(n), val_indices)
+
+    # Get the underlying table
+    table = dataset.data
+
+    # Create separate tables for each split
+    train_table = table.filter(pa.array(train_mask))
+    test_table = table.filter(pa.array(test_mask))
+    val_table = table.filter(pa.array(val_mask))
```

Comment:
What difference would it make to use [select](https://huggingface.co/docs/datasets/v1.8.0/package_reference/main_classes.html#datasets.Dataset.select)?
According to what I saw in the documentation, you can pass the indexes directly, making the code a little more readable.

Reply:
Hi! I didn't code this section originally, I tried on using select but I found some problems implementing it,  Select and Filter method from Dataset give a new Dataset with the selected rows but is doesn't update the original table (so the test returns error). I think this is why we are filtering the Table not the Dataset. When I try to use the select or filter method with the Table instaed of the Dataset (passing only the indices as parameters) it throws IndexErrors because of indx out of bounds.

I'm currently trying to track the error as the indices I'm giving to select rows seems to be ok.

Reply:
In this case it is different because the dataset is supposed to be already saved on disk. 
So, if you modify the dataset in memory and save it in another file, you would not be changing the original one.